cmd/backup/main.go

```diff
@@ -330,10 +331,21 @@ func (s *script) stopContainers() (func() error, error) {
 // saves it to disk.
 func (s *script) takeBackup() error {
 	s.file = timeutil.Strftime(&s.start, s.file)
-	if err := targz.Compress(s.c.BackupSources, s.file); err != nil {
+  backupCopiedSource := filepath.Join("/tmp", s.c.BackupSources)
+	// copy before compressing guard against a situation where backup folder's content are still growing.
+	if err := copy.Copy(s.c.BackupSources, backupCopiedSource); err != nil {
```

Comment:
Two questions about the behavior of this library:

- what happens if there are symlinks in the copied file tree? Do they get resolved?
- what happens if files in `s.c.BackupSources` change while copying is done? (i.e. how does this not fail in such a scenario)

Reply:
1. no symlinks are not copied by default (https://github.com/otiai10/copy/blob/edeb9f37d4d0c28725014fcab6635120197932a1/copy.go#L163).
2. since we are compressing the snapshot folder (copied folder) now, so even if the content of s.c.BackupSources is grwoing, it does not affect the snapshot folder (copied folder). So the archived content is the content in the moment copy finish. anything added after that is not archived.

Reply:
> no symlinks are not copied by default

From reading the code it seems to me `Shallow` is used by default, but that's good as it would be the exact same behavior as before: https://github.com/otiai10/copy/blob/edeb9f37d4d0c28725014fcab6635120197932a1/options.go#L70-L72

Reply:
I also wonder if we should use the `PreserveTimes` option here as otherwise all of the copied files would lose their timestamps (which might be important for some consumers): https://github.com/otiai10/copy/tree/edeb9f37d4d0c28725014fcab6635120197932a1#advanced-usage

Reply:
I see, will add that!